config/config_test.go

```diff
@@ -70,3 +73,115 @@ func TestDetermineUseLiveData(t *testing.T) {
 		test.That(t, useLiveData, test.ShouldBeTrue)
 	})
 }
+
+// makeCfgService creates the simplest possible config that can pass validation.
+func makeCfgService() config.Service {
+	model := resource.NewDefaultModel(resource.ModelName("test"))
+	cfgService := config.Service{Name: "test", Type: "slam", Model: model}
+	cfgService.Attributes = make(map[string]interface{})
+	cfgService.Attributes["config_params"] = map[string]string{
+		"mode": "test mode",
+	}
+	cfgService.Attributes["data_dir"] = "path"
+	cfgService.Attributes["use_live_data"] = true
+	cfgService.Attributes["delete_processed_data"] = true
```

Comment:
delete_processed_data isn't a required parameter, so it shouldn't be included in the simplest possible config.